Add flag "--pidfile" for podman create/run (#10041)
containers/podman: docs/source/markdown/podman-create.1.md

The image developer can set a different default with the WORKDIR instruction. The operator
can override the working directory by using the **-w** option.

#### **\-\-pidfile**=*path*

Write the pid of the container process to a file.

> The default pidfile is RunDir/pidfile.

Reviewer: I had a hard time translating this to the location of the pidfile.  Since we don't document the location of the conmon pidfile, I am not sure this is helpful to the user?  When you set this flag, does it store the pidfile in two locations?

Reviewer: After this is done, will I be able to inspect the location of the pidfile?

 podman inspect -l  | grep -i pidfile
        "ConmonPidFile": "/run/user/3267/containers/overlay-containers/7e0e1fadac3f047aa7c880c694a93186768feab9a2198d0f620a01706094b428/userdata/conmon.pid",

Reviewer: If you are hiding this from remote, then it should be documented that it is not supported for remote.

Author: I write this docoment according to https://github.com/containers/podman/blob/master/docs/source/markdown/podman-create.1.md#--conmon-pidfilepath.
When set this flag, it stores the pidfile in one location.

Author: After this is done, the container‘s inspection will show the pidfile location in CreateCommand, not a property. Is it ok?

podman inspect -l | grep -i pidfile
"CreateCommand": [
                "podman",
                "create",
                "--pidfile",
                "/root/a.txt",

Reviewer: Understood, but this line:

 The default pidfile is RunDir/pidfile. 

Is not really helpful, or at least forced me to do detectivework.  As a user, I have no idea where RunDir is?

Something like: to find the default location use:
```
You can find the default location for this file by executing the following command on the contaner.
$ podman inspect --format '{{ .ConmonPidFile }}' $CID
/run/user/1000/containers/overlay-containers/7e0e1fadac3f047aa7c880c694a93186768feab9a2198d0f620a01706094b428/userdata/conmon.pid
```

Reviewer: You need to add the Inspection of PidFile to match conmonpidfile,

Author: Added detail description for pidfile. Added description about how to find the pidfile location inspect command.
And documented the options conmon-pidfile and pidfile are not support on remote.

The image developer can set a different default with the WORKDIR instruction. The operator
can override the working directory by using the **-w** option.

#### **\-\-pidfile**=*path*

> When specify the pidfile location, container process pid will be written this pidfile.(This option is not available with the remote Podman client)

Reviewer 2: ```suggestion
When the pidfile location is specified, the container process' PID will be written to the pidfile. (This option is not available with the remote Podman client)
```

> After container started, you can find the location for this file by the following command on the contaner.

Reviewer 2: ```suggestion
After the container is started, the location for the pidfile can be discovered with the following `podman inspect` command:
```

    $ podman inspect --format '{{ .Config.CreateCommand }}' $CID
    podman run --pidfile $pidfilepath image

> If not specify this option, container process pid will be written to

Reviewer 2: ```suggestion
If the pidfile option is not specified, the container process' PID will be written to
```

/run/containers/storage/${storage-driver}-containers/$CID/userdata/pidfile.

> The default pidfile location is not in container inspection.

Reviewer 2: ```suggestion
The default pidfile location is not listed in the `podman inspect` output.
```

> The default pidfile location is not listed in the `podman inspect` output.

Reviewer: No we need you to add the pid file location to the inspect.  Just copy the way that conmonpidfile is handled.

Author: Sorry for misunderstanding.
